recursio-client/Characters/PlayerInput.gd

```diff
@@ -37,14 +36,22 @@ func _physics_process(_delta):
 		InputManager.add_rotation_to_input_frame(rotate_vector)
 
 		var buttons_pressed: int = _get_buttons_pressed()
+		# invert pressed buttons for shooting here
+		# to delay ghosts triggering actions to button_released
+		if buttons_pressed & ActionManager.Trigger.FIRE_START:
+			_player.aim_mode = true
+			buttons_pressed &= ~(ActionManager.Trigger.FIRE_START)
+		elif _player.aim_mode == true:
+			_player.aim_mode = false
+			buttons_pressed |= ActionManager.Trigger.FIRE_START
+
 		_player.apply_input(movement_vector, rotate_vector, buttons_pressed)
 		InputManager.set_triggers_in_input_frame(buttons_pressed)
-
 		InputManager.close_current_input_frame()
 		InputManager.send_player_input_data_to_server()
 	elif cur_phase == RoundManager.Phases.PREPARATION:
-		if Input.is_action_just_pressed(_player_timeline_pick_trigger):
-			var timeline_index: int = (_player.timeline_index + 1) % (Constants.get_value("ghosts","max_amount") + 1)
+		if Input.is_action_just_pressed("player_switch"):
```

Comment:
Is there a reason for replacing this with the string directly?

Reply:
joah, weil das eh nur einmal verwendet wurde und sowieso eine quasi-constant is, weil die direkt aus den input mappings vom project kommt

Reply:
Alle anderen sind halt auch oben definiert, daher würde ich das entweder damit kombinieren oder so lassen. Dann ist's einheitlich und wenn es sich ändert, muss man es nur oben suchen (my opinion)

Reply:
kein plan, bin nur über das eine gestolpert und dachte mir, das is eh in den project settings, also warum doppelt?
Außer man mag nicht direkt mit strings hantieren, was eigentlich eh schirch is aber halt godot-comon,
aber dann wär ne allgemeine Stelle für alle input keys sowieso auch gscheiter als nur im PlayerInput

Reply:
Willst das Letztere gleich machen? \:D